CONTRIBUTING.md

```diff
@@ -37,27 +37,29 @@ to avoid duplication of effort.
 The process of submitting a Pull Request is relatively simple, following a 
 [rebase workflow](http://kensheedlo.com/essays/why-you-should-use-a-rebase-workflow/).  Please use the
 following pattern:
+
 1. Create a new branch for your fix/feature
 1. Make your changes
-1. Rebase onto the upstream
-1. Run the test suite
-1. Squash your commits
+1. Run the test suite *(if failed, goto 2)*
+1. Squash your commits *(re-run test suite)*
+1. Rebase onto the upstream *(re-run test suite)*
```

Comment:
I put the `rebase` step ahead of running the tests in order to encourage testing after fetching the latest changes from the repo.  I think that the original order would be better suited, as it checks the changes against the latest from upstream, not what the author had in their branch to start with.